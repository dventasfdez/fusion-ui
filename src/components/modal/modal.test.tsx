import React, {useState} from 'react';
import Modal from './modal';
import ModalBody from './modalBody';
import ModalFooter from './modalFooter';
import ModalHeader from './modalHeader';
import {fireEvent, render} from '@testing-library/react';
import renderer from 'react-test-renderer';

const ModalTest: React.FC = () => {
  const [modalOpen, setModalOpen] = useState(false);
  return (
    <div id="root">
      <button data-testid="modal-show-btn" onClick={() => setModalOpen(true)}></button>
      <Modal
        data-testid="modal-test"
        className=.stepone-ui"
        open={modalOpen}
        onClose={() => setModalOpen(false)}
        hasIconClose={true}
        renderAsPortal
      >
        <ModalHeader>My Title</ModalHeader>
        <ModalBody>
          Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odio aperiam libero hic dolorum veritatis,
          necessitatibus, neque suscipit autem nostrum repellat corporis nihil, a saepe est. Autem beatae sunt debitis quasi!
        </ModalBody>
        <ModalFooter>
          <button>Close Button</button>
        </ModalFooter>
      </Modal>
    </div>
  );
};
const ModalWithoutTestId = () => (
  <Modal>
    <ModalHeader>My Title</ModalHeader>
    <ModalBody>
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odio aperiam libero hic dolorum veritatis, necessitatibus,
      neque suscipit autem nostrum repellat corporis nihil, a saepe est. Autem beatae sunt debitis quasi!
    </ModalBody>
    <ModalFooter>
      <button>Close Button</button>
    </ModalFooter>
  </Modal>
);

it('render Modal match snap without test ids', () => {
  const component = renderer.create(<ModalWithoutTestId />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Modal render when click button and close when click icon close ', () => {
  const {container, getByTestId} = render(<ModalTest />);
  const modalBtn = getByTestId('modal-show-btn');
  if (modalBtn) fireEvent.click(modalBtn);
  expect(getByTestId('modal-test')).toBeDefined();
  const closeModalBtn = getByTestId('modal-test-icon-close');
  if (closeModalBtn) fireEvent.click(closeModalBtn);
  expect(container.getElementsByClassName('tooltip').length).toBe(0);
});
