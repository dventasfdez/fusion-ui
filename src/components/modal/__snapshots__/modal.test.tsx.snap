// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Modal match snap without test ids 1`] = `
<div
  className="modal "
>
  <div
    className="modal-header "
  >
    My Title
  </div>
  <div
    className="modal-body "
  >
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odio aperiam libero hic dolorum veritatis, necessitatibus, neque suscipit autem nostrum repellat corporis nihil, a saepe est. Autem beatae sunt debitis quasi!
  </div>
  <div
    className="modal-footer "
  >
    <button>
      Close Button
    </button>
  </div>
</div>
`;
